### ISSUE TITLE
Reporting problems with custom parser

### ISSUE THREAD

**Issue by awinogradov** [postcss/postcss-reporter#11]
If i use reporter with custom parser without extending postcss parser, i see this:
`Unhandled rejection TypeError: Cannot read property 'input' of undefined`. Yes, i build AST with API and i don't create Input object. I return postcss.root object to custom parser option.

**Comment by davidtheclark** [comment]
Am I right in thinking that this is the line causing the error: https://github.com/postcss/postcss-reporter/blob/f617f2a4977c7580a08dc4de6623a512e439f3ad/lib/reporter.js#L22? Do you have a stack trace?

**Comment by awinogradov** [comment]
```
Unhandled rejection TypeError: Cannot read property 'input' of undefined
    at /Users/verybigman/Sources/parser/node_modules/postcss-reporter/lib/reporter.js:22:33
    at LazyResult.run (/Users/verybigman/Sources/parser/node_modules/postcss/lib/lazy-result.js:201:20)
    at LazyResult.sync (/Users/verybigman/Sources/parser/node_modules/postcss/lib/lazy-result.js:188:32)
    at LazyResult.stringify (/Users/verybigman/Sources/parser/node_modules/postcss/lib/lazy-result.js:212:14)
    at LazyResult.get (/Users/verybigman/Sources/parser/node_modules/postcss/lib/lazy-result.js:241:25)
    at Transformer.css (/Users/verybigman/Sources/parser/lib/transformer.js:46:39)
    at /Users/verybigman/Sources/parser/lib/index.js:27:31
    at tryCatcher (/Users/verybigman/Sources/parser/node_modules/bluebird/js/main/util.js:26:23)
    at Promise._settlePromiseFromHandler (/Users/verybigman/Sources/parser/node_modules/bluebird/js/main/promise.js:503:31)
    at Promise._settlePromiseAt (/Users/verybigman/Sources/parser/node_modules/bluebird/js/main/promise.js:577:18)
    at Promise._settlePromises (/Users/verybigman/Sources/parser/node_modules/bluebird/js/main/promise.js:693:14)
    at Async._drainQueue (/Users/verybigman/Sources/parser/node_modules/bluebird/js/main/async.js:123:16)
    at Async._drainQueues (/Users/verybigman/Sources/parser/node_modules/bluebird/js/main/async.js:133:10)
    at Immediate.Async.drainQueues [as _onImmediate] (/Users/verybigman/Sources/parser/node_modules/bluebird/js/main/async.js:15:14)
    at processImmediate [as _immediateCallback] (timers.js:371:17)
```

**Comment by davidtheclark** [comment]
@awinogradov : This should be fixed as of 64951005aa50f9e085bf27812cf3587164385294 and release 1.2.1. Is it working for you now?

**Comment by davidtheclark** [comment]
I believe this is fixed because it is work for stylelint. Closing. Please re-open if you or anybody else runs into the problem again.
